test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,92 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_sequence) {
+  rmw_ret_t ret;
+  size_t count = 1u;
+  size_t taken;
```

Comment:
I think taken is supposed to be a boolean (guessing on the name)? Also, initialize it and I suggest you to move it to line 393 prior the call to `rmw_take_sequence`
```suggestion
  bool taken = false;
```

Reply:
Acording to API: 
```
rmw_take_sequence( const rmw_subscription_t * subscription,	size_t count, rmw_message_sequence_t * message_sequence, rmw_message_info_sequence_t * message_info_sequence, size_t * taken, rmw_subscription_allocation_t * allocation);
```

Reply:
Just checked, you are right, is not a boolean. nit: init to zero, and check expected value after call to `rmw_take_sequence`

Reply:
Updated on dc8ab77